Framework/API/inc/MantidAPI/AlgorithmFactory.h

```diff
@@ -111,8 +112,8 @@ class MANTID_API_DLL AlgorithmFactoryImpl final : public Kernel::DynamicFactory<
   const std::vector<std::string> getKeys() const override;
   const std::vector<std::string> getKeys(bool includeHidden) const;
 
-  /// Get an algorithms name from the alias map
-  boost::optional<std::string> getRealNameFromAlias(const std::string &alias) const noexcept;
+  /// Get an algorithms name and version from the alias map
+  boost::optional<std::pair<std::string, int>> getRealNameFromAlias(const std::string &alias) const noexcept;
```

Comment:
Would it be possible to modernise this code by using `std::optional` instead of `boost::optional`?